src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java

```diff
@@ -368,6 +382,18 @@ private String createDeployment(AWSClients aws, RevisionLocation revisionLocatio
         return createDeploymentResult.getDeploymentId();
     }
 
+    private String stopDeployment(AWSClients aws, String deploymentId) throws Exception {
+
+        this.logger.println("Stopping deployment " + deploymentId);
+
+        StopDeploymentResult stopDeploymentResult = aws.codedeploy.stopDeployment(
```

Comment:
Stopping a deployment might not be immediate. We might want to wait for the deployment to actually stop before continuing.